calicoctl/commands/version.go

```diff
@@ -58,27 +61,18 @@ Description:
 	fmt.Println("Git commit:       ", GIT_REVISION)
 
 	// Load the client config and connect.
-	//cf := parsedArgs["--config"].(string)
-	//client, err := clientmgr.NewClient(cf)
-	//if err != nil {
-	//	fmt.Println(err)
-	//	os.Exit(1)
-	//}
-	//cfg := client.Config()
-	//a := apiconfig.NewCalicoAPIConfig()
-	//
-	//val, assigned, err := cfg.GetFelixConfig("CalicoVersion", "")
-	//if err != nil {
-	//	val = fmt.Sprintf("unknown (%s)", err)
-	//} else if !assigned {
-	//	val = "unknown"
-	//}
-	//fmt.Println("Cluster Version:  ", val)
-	//val, assigned, err = cfg.GetFelixConfig("ClusterType", "")
-	//if err != nil {
-	//	val = fmt.Sprintf("unknown (%s)", err)
-	//} else if !assigned {
-	//	val = "unknown"
-	//}
-	//fmt.Println("Cluster Type:     ", val)
+	cf := parsedArgs["--config"].(string)
+	client, err := clientmgr.NewClient(cf)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	ctx := context.Background()
+	ci, err := client.ClusterInformation().Get(ctx, "default", options.GetOptions{})
+	if err != nil {
+		fmt.Println("Unable to retrieve Cluster Version or Type: ", err)
+	} else {
+		fmt.Println("Cluster Version:  ", ci.Spec.CalicoVersion)
```

Comment:
We need to check if the values are assigned manually for both `ClusterVersion` and `ClusterType` print `"unknown"` if it's `""` like we use to before.

Reply:
Is that what the previous code equated to? IIRC, previously the "" is what was returned as val if there was not a value set, so you could have "" in the fields but `assigned` would still be true and nothing would be printed.  I'll go check.

Reply:
Ok so the "" before was the node.
Previously if the `value` in the GetFelixConfig was nil then assigned would be false, otherwise true (it could be set empty).
So I don't know if it matters but it doesn't look like we really had this specifically as a case before. I feel like the fact that the Get on the ClusterInformation would give an error of `cerrors.ErrorResourceDoesNotExist` would equate to the previous check if the Config field was assigned or not. WDYT?

Reply:
Now that we have a bunch of fields aggregated into a single resource `ClusterInformation` we can have some of the fields un-assigned (ClusterType) even when the `default` ClusterInformation resource exists, and the only way to know is to see if it's a `""` now since they're all strings